plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/ExMemory.hs

```diff
@@ -206,7 +206,7 @@ instance ExMemoryUsage Data where
         where sizeData d =
                   nodeMem +
                      case d of
-                       Constr _ l -> sizeDataList l
+                       Constr n l -> memoryUsage n +  sizeDataList l
```

Comment:
This will change the costing of existing scripts, right?

Reply:
Do you need this for costing `serializeData`?

Reply:
Damn. @bezirg pointed out that it should have taken account of the size of the tag so I just added it without it occurring to me that that would change the costs of old scripts. Maybe we should change it back. In practice the tags are surely never going to be more than 64 bits long anyway.   Would it be possible to cheaply construct a massive tag at runtime that would cause excessive CPU consumption?

Reply:
Probably not. Also I think most of the `data` functions are currently constant-cost, so maybe this literally changes nothing?

Reply:
Hah! they are indeed all constant cost. So we can do what we like.

Reply:
> Hah! they are indeed all constant cost. So we can do what we like.

`equalsData`?

Reply:
Damn, yes.

Reply:
I think we'll have to revert that then.  With that particular cost function it would be OK if the size decreased, but this change could definitely make things cost more.  Another reason to have configurable size functions.

Reply:
Okay, can you make a ticket for fixing this, then? Blocked by the configurable size functions issue.

Reply:
> Okay, can you make a ticket for fixing this, then? Blocked by the configurable size functions issue.

See SCP-3677.